packages/api/src/trpc.ts

```diff
@@ -101,7 +113,19 @@ export const publicProcedure = t.procedure;
  * TODO: Add middleware for authentication
  *
  */
-export const protectedProcedure = t.procedure;
+
+const isAuthed = t.middleware(({ next, ctx }) => {
+  if (!ctx.auth.userId) {
+    throw new TRPCError({ code: "UNAUTHORIZED" });
+  }
+  return next({
+    ctx: {
+      auth: ctx.auth,
+    },
+  });
+});
+
+export const protectedProcedure = t.procedure.use(isAuthed);
```

Comment:
can you make `adminProcedure` equal to `protectedProcedure` (we can add another middleware to it later)